packages/flutter/lib/src/material/menu_bar.dart

```diff
@@ -0,0 +1,3396 @@
+                    if (_moveToPreviousTopLevel(currentMenu)) {
+                      return;
+                    }
+                    break;
+                  case Axis.vertical:
+                    if (buttonIsFocused) {
+                      if (_moveToPreviousTopLevel(currentMenu)) {
+                        return;
+                      }
+                    } else {
+                      if (_moveToParent(currentMenu)) {
+                        return;
+                      }
+                    }
+                    break;
+                }
+                break;
+            }
+            break;
+        }
+        break;
+      case TraversalDirection.right:
+        switch (orientation) {
+          case Axis.horizontal:
+            switch (Directionality.of(context)) {
+              case TextDirection.rtl:
+                if (_moveToPrevious(currentMenu)) {
+                  return;
+                }
+                break;
+              case TextDirection.ltr:
+                if (_moveToNext(currentMenu)) {
+                  return;
+                }
+                break;
+            }
+            break;
+          case Axis.vertical:
+            switch (Directionality.of(context)) {
+              case TextDirection.rtl:
+                switch (currentMenu.parent.orientation) {
+                  case Axis.horizontal:
+                    if (_moveToPreviousTopLevel(currentMenu)) {
+                      return;
+                    }
+                    break;
+                  case Axis.vertical:
+                    if (_moveToParent(currentMenu)) {
+                      return;
+                    }
+                    break;
+                }
+                break;
+              case TextDirection.ltr:
+                if (buttonIsFocused) {
+                  if (_moveToSubmenu(currentMenu)) {
+                    return;
+                  }
+                } else {
+                  if (_moveToNextTopLevel(currentMenu)) {
+                    return;
+                  }
+                }
+                break;
+            }
+            break;
+        }
+        break;
+    }
+    super.invoke(intent);
+  }
+}
+
+// Wraps the MaterialStateMouseCursor so that it can default to
+// MouseCursor.uncontrolled if none is set, and so it has a debug description.
+class _MouseCursor extends MaterialStateMouseCursor {
+  const _MouseCursor(this.resolveCallback);
+
+  final MaterialPropertyResolver<MouseCursor?> resolveCallback;
+
+  @override
+  MouseCursor resolve(Set<MaterialState> states) => resolveCallback(states) ?? MouseCursor.uncontrolled;
+
+  @override
+  String get debugDescription => 'Menu_MouseCursor';
+}
+
+bool _menuDebug(String message, [Iterable<String>? details]) {
+  if (_kDebugMenus) {
+    debugPrint('MENU: $message');
+    if (details != null && details.isNotEmpty) {
+      for (final String detail in details) {
+        debugPrint('    $detail');
+      }
+    }
+  }
```

Comment:
wrap this in an assert so it is guaranteed to get removed in release builds?

Reply:
Each usage is already wrapped, but for insurance, sure.

(I could just assert we're not in release mode, right? LOL! (j/k))